mysql/mysql.go

```diff
@@ -0,0 +1,201 @@
+package mysql
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	// import mysql driver anonymously (just run the init)
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/pkg/errors"
+	nosql "github.com/smallstep/nosql/database"
+)
+
+// DB is a wrapper over *sql.DB,
+type DB struct {
+	db *sql.DB
+}
+
+// Open creates a Driver and connects to the database with the given address
+// and access details.
+func (db *DB) Open(dataSourceName string, opt nosql.Options) error {
+	var err error
+	_db, err := sql.Open("mysql", dataSourceName)
+	if err != nil {
+		return errors.Wrap(err, "error connecting to mysql")
+	}
+	_, err = _db.Exec("CREATE DATABASE IF NOT EXISTS test")
```

Comment:
Why do we create a test database, in mysql is very common to use a dataSourceName with the db on it `user:password@/dbname`

Reply:
that was an accident. Forgot to fix that. Mean to create the database stored in the options (the reason being that it may not have been created yet).

Reply:
and to your point, yea normally people have the DB name on the datasourcename, but that Im pretty sure that fails if the database doesn't exist.